### PR TITLE
add support for multiple conditions

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/67143

Born in the context of the ticket this PR adds support for multiple conditions.  
By default the relation between the conditions will be the logic `or`:

```html
<span
    class="tribe-dependent"
    data-depends="#select"
    data-condition-empty
    data-condition="Foo"
>
```

The `span` **will show** if `#select` is empty OR has a value of 'Foo'.
To change the condition relation to the logic `and` one can use the `data-condition-relation="and"` attribute.

```html
<span
    class="tribe-dependent"
    data-depends="#select"
    data-condition-relation="and"
    data-condition-not-empty
    data-condition-not="Foo"
>
```

For the `span` **to show** `#select` should be not empty AND not have a value of 'Foo'.  
I'm setting the default condition to the logic `or` to follow along WordPress standard.

### CODE REVIEW

File: src/resources/js/dependency.js [hunk 1]
```diff
@@ -44,26 +70,35 @@
...
+						return is_applicable;
+					} );
+
+					if ( condition_relation === 'and' ) {
+						passes = _.reduce( constraints, function ( passes, constraint, key ) {
+							return passes && constraint_conditions[key]( value, constraint, $field );
+						}, true );
+					} else {
+						passes = _.reduce( constraints, function ( passes, constraint, key ) {
+							return passes || constraint_conditions[key]( value, constraint, $field );
```
Comment (bordoni) [inline src/resources/js/dependency.js]: spacing around `[key]`

File: src/resources/js/dependency.js [hunk 2]
```diff
@@ -44,26 +70,35 @@
...
+						condition_relation = $dependent.data( 'relation' ) || 'and',
+						passes;
+
+					constraints = _.pick( constraints, function ( is_applicable ) {
+						return is_applicable;
+					} );
+
+					if ( condition_relation === 'and' ) {
+						passes = _.reduce( constraints, function ( passes, constraint, key ) {
+							return passes && constraint_conditions[key]( value, constraint, $field );
```
Comment (bordoni) [inline src/resources/js/dependency.js]: spacing around `[key]`

File: src/resources/js/dependency.js [hunk 3]
```diff
@@ -44,26 +70,35 @@
...
+						}, true );
+					} else {
+						passes = _.reduce( constraints, function ( passes, constraint, key ) {
+							return passes || constraint_conditions[key]( value, constraint, $field );
+						}, false );
+					}
+
+					if ( passes && !is_disabled ) {
```
Comment (bordoni) [inline src/resources/js/dependency.js]: Spacing after `!`

**Comment by lucatume** [comment]
changes in

**Comment by bordoni** [comment]
👍

**Pull request merged**